pkg/graph/resolvers/system_intake_status_requester.go

```diff
@@ -0,0 +1,107 @@
+	}
+}
+
+func calcSystemIntakeInitialFormStatusRequester(intakeFormState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch intakeFormState {
+	case models.SIRFSNotStarted:
+		return models.SISRInitialRequestFormNew, nil
+	case models.SIRFSInProgress:
+		return models.SISRInitialRequestFormInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRInitialRequestFormEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRInitialRequestFormSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid intakeFormState")
+	}
+}
+
+func calcSystemIntakeDraftBusinessCaseStatusRequester(draftBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch draftBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
+		return models.SISRDraftBusinessCaseInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRDraftBusinessCaseEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRDraftBusinessCaseSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid draftBusinessCaseState")
+	}
+}
+
+func calcSystemIntakeGRTMeetingStatusRequester(grtDate *time.Time, currentTime time.Time) models.SystemIntakeStatusRequester {
+	if grtDate == nil || grtDate.After(currentTime) {
+		return models.SISRGrtMeetingReady
+	}
+
+	return models.SISRGrtMeetingAwaitingDecision
+}
+
+func calcSystemIntakeFinalBusinessCaseStatusRequester(finalBusinessCaseState models.SystemIntakeFormState) (models.SystemIntakeStatusRequester, error) {
+	switch finalBusinessCaseState {
+	case models.SIRFSNotStarted, models.SIRFSInProgress:
+		return models.SISRFinalBusinessCaseInProgress, nil
+	case models.SIRFSEditsRequested:
+		return models.SISRFinalBusinessCaseEditsRequested, nil
+	case models.SIRFSSubmitted:
+		return models.SISRFinalBusinessCaseSubmitted, nil
+	default:
+		return "", fmt.Errorf("issue calculating the requester intake status, no valid finalBusinessCaseState")
+	}
+}
+
+func calcSystemIntakeGRBMeetingStatusRequester(grbDate *time.Time, currentTime time.Time) models.SystemIntakeStatusRequester {
+	if grbDate == nil || grbDate.After(currentTime) {
+		return models.SISRGrbMeetingReady
+	}
+
+	return models.SISRGrbMeetingAwaitingDecision
+}
+
+func calcSystemIntakeDecisionStatusRequester(decisionState models.SystemIntakeDecisionState) (models.SystemIntakeStatusRequester, error) {
+	switch decisionState {
+	case models.SIDSLcidIssued:
+		return models.SISRLcidIssued, nil
+	case models.SIDSNotApproved:
+		return models.SISRNotApproved, nil
+	case models.SIDSNotGovernance:
+		return models.SISRNotGovernance, nil
+	case models.SIDSNoDecision:
+		return "", fmt.Errorf("issue calculating the requester intake status - invalid state, intake decisionState is NO_DECISION, but intake is not closed")
```

Comment:
I like this error message being super clear about what happened. I had suggested to Steven possibly putting something like this at the top of the file to make it very clear that we are checking for a specific invalid state, but I think this is a good way to handle it as well.

```
if (
  intake.Step ==  models.SystemIntakeStepDECISION &&
  intake.DecisionState == models.SIDSNoDecision
) return "", fmt.Errorf("invalid state...")
// switch
```

Reply:
Yeah, at this point in the code we know _exactly_ what's invalid, so I figured we could have a pretty specific error message to help us debug if necessary.